src/protocols/BaseProtocolAdapter.ts

```diff
@@ -96,6 +97,19 @@ export abstract class BaseProtocolAdapter {
 
   abstract fetchAccountTokenBalances(address: string): Promise<ITokenBalance[]>;
 
+  abstract transferToken(
+    amount: number,
+    recipient: string,
+    contractId: string,
+    options: {
+      fromAccount: IAccount;
+      maxPriorityFeePerGas: string;
+      maxFeePerGas: string;
+    },
+  ): Promise<{ hash: string }>;
```

Comment:
The type `{ hash: string }` is already used in some places. I suggest to create separate transfer response type and reuse it.